src/vtvalidate.cpp

```diff
@@ -5,55 +5,83 @@
 #include <iostream>
 #include <map>
 #include <stdexcept>
-
-/**
- * This is an asynchronous standalone function that logs a string.
- * @name helloAsync
- * @param {Object} args - different ways to alter the string
- * @param {boolean} args.louder - adds exclamation points to the string
- * @param {Function} callback - from whence the hello comes, returns a string
- * @returns {string}
- * @example
- * var module = require('./path/to/lib/index.js');
- * module.helloAsync({ louder: true }, function(err, result) {
- *   if (err) throw err;
- *   console.log(result); // => "...threads are busy async bees...hello
- * world!!!!"
- * });
- */
+#include <vtzero/vector_tile.hpp>
 
 namespace VectorTileValidate {
 
-// Expensive allocation of std::map, querying, and string comparison,
-// therefore threads are busy
-std::string do_expensive_work(bool louder) {
-
-    std::map<std::size_t, std::string> container;
-    std::size_t work_to_do = 100000;
-
-    for (std::size_t i = 0; i < work_to_do; ++i) {
-        container.emplace(i, std::to_string(i));
-    }
-
-    for (std::size_t i = 0; i < work_to_do; ++i) {
-        std::string const& item = container[i];
-        if (item != std::to_string(i)) {
-
-            // AsyncHelloWorker's Execute function will take care of this error
-            // and return it to js-world via callback
-            // Marked NOLINT to avoid clang-tidy cert-err60-cpp error which we cannot
-            // avoid on some linux distros where std::runtime_error is not properly
-            // marked noexcept. Details at https://www.securecoding.cert.org/confluence/display/cplusplus/ERR60-CPP.+Exception+objects+must+be+nothrow+copy+constructible
-            throw std::runtime_error("Uh oh, this should never happen"); // NOLINT
+struct geom_handler {
+
+    void points_begin(uint32_t /*dummy*/) {}
+
+    void points_point(const vtzero::point /*dummy*/) {}
+
+    void points_end() const noexcept {}
+
+    void linestring_begin(uint32_t /*dummy*/) {}
+
+    void linestring_point(const vtzero::point /*dummy*/) {}
+
+    void linestring_end() const noexcept {}
+
+    void ring_begin(uint32_t /*dummy*/) {}
+
+    void ring_point(const vtzero::point /*dummy*/) {}
+
+    void ring_end(vtzero::ring_type /*dummy*/) const noexcept {}
+
+}; // struct geom_handler
+
+std::string parseTile(vtzero::data_view const& buffer) {
+    std::string result;
+
+    vtzero::vector_tile tile{buffer};
+    try {
+        while (auto layer = tile.next_layer()) {
+            while (auto feature = layer.next_feature()) {
+                // Detect geomtype of feature and decode
+                geom_handler handler;
+                vtzero::decode_geometry(feature.geometry(), handler);
+
+                feature.reset_property(); // restart the property iterator from the beginning
```

Comment:
I think this `feature.reset_property();` is unneeded because there is not another operation beforehand that had modified the properties.